python/pyspark/sql/readwriter.py

```diff
@@ -431,6 +432,8 @@ def jdbc(self, url, table, column=None, lowerBound=None, upperBound=None, numPar
         if column is not None:
             if numPartitions is None:
                 numPartitions = self._spark._sc.defaultParallelism
+            assert lowerBound != None, "lowerBound can not be None when ``column`` is specified"
+            assert upperBound != None, "upperBound can not be None when ``column`` is specified"
```

Comment:
Yes, The Scala code could check this, but the PySpark code will fail at ```int(lowerBound)``` first, so the customer is confused.